### PR TITLE
Add `compose exec` command

### PR DESCRIPTION
**Pull request by djdongjin**
Implement `compose exec` command. docker doc: https://docs.docker.com/engine/reference/commandline/compose_exec/

Also help feature request from downstream community that relies on nerdctl (finch: https://github.com/runfinch/finch/issues/79)

Signed-off-by: Jin Dong <jindon@amazon.com>

### CODE REVIEW

**Comment by djdongjin** [comment]
While implementing `compose exec`, I notice the `nerdctl exec` command mentions "`-t, --tty` should correspond to `-i`":

https://github.com/containerd/nerdctl/blob/dff398c6a9ca40db8f9c25c70e76455267564f51/cmd/nerdctl/exec.go#L53

However, I didn't see this is checked, and they are more independent:

https://github.com/containerd/nerdctl/blob/dff398c6a9ca40db8f9c25c70e76455267564f51/cmd/nerdctl/exec.go#L143-L149

https://github.com/containerd/nerdctl/blob/a8ab5cdb3dda7cb64f9e4f964105243b4ab5ef27/cmd/nerdctl/exec_linux_test.go#L69

Is this restriction removed? I saw similar code in `run` and `compose run`, and `compose run` is following the document:

https://github.com/containerd/nerdctl/blob/7276e4e3ef3b6fddcb7654c5188e40f1ba34fa85/cmd/nerdctl/compose_run.go#L119

File: pkg/composer/exec.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	args = append(args, eo.Args...)
+	cmd := c.createNerdctlCmd(ctx, args...)
+
+	if eo.Interactive {
+		cmd.Stdin = os.Stdin
+	}
+	if !eo.Detach {
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+	}
```
Comment (djdongjin) [inline pkg/composer/exec.go:89]: Not sure if this IO setup is too simplified, still looking at how other commands (`compose run`, `run`, `exec`) handle IO (given `-i`/`-d`/`-t`). Appreciate any suggestion :)

**Comment by djdongjin** [comment]
> > Is this restriction removed?
> 
> Yes, probably

Seems fixed in this commit: https://github.com/containerd/nerdctl/commit/18ea81f8f7cb6949b4748349b2782aad19e4d484

I'll take a look!

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: cmd/nerdctl/compose_exec_linux_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		"nobody:users": "uid=65534(nobody) gid=100(users)",
+	}
+
+	for userStr, expected := range testCases {
+		args := []string{"-f", comp.YAMLFullPath(), "exec", "-i=false", "-t=false"}
+		if userStr != "" {
+			args = append(args, "--user", userStr)
+		}
+		args = append(args, "svc0", "id")
+		base.ComposeCmd(args...).AssertOutContains(expected)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/compose_exec_linux_test.go:94]: AssertOutExactly might be preferable

**Comment by AkihiroSuda** [comment]
Merged, but please take a look: https://github.com/containerd/nerdctl/pull/1598#discussion_r1040579720

File: cmd/nerdctl/compose_exec_linux_test.go (hunk 2 above)
Comment (djdongjin) [reply cmd/nerdctl/compose_exec_linux_test.go:94]: Thanks for the suggestion.

`id` output (the empty case) has extra strings:

```
➜  compose-exec sndc exec svc0 id
uid=0(root) gid=0(root) groups=1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)
➜  compose-exec sndc exec -u 1000 svc0 id
uid=1000 gid=0(root)
```

**Pull request merged**